### ISSUE TITLE
Cannot set property '__disabled' of undefined after zoomed chart data update

### ISSUE THREAD

**Issue by lticadler** [amcharts/amcharts4#2693]
Unfortunately this problem is not so easy to fill into a pen :/
Hopefully my explanation with the stack is enough.

- XYChart with DateAxis with one LineSeries. Data is set per series not per chart.
- 4 days of data (not filled regularly)
![image](https://user-images.githubusercontent.com/58069494/87297702-28e9a600-c509-11ea-9b17-3811291a2ce3.png)

- zoom into the chart
![image](https://user-images.githubusercontent.com/58069494/87297626-09eb1400-c509-11ea-88bf-7b074f6ebce9.png)

![image](https://user-images.githubusercontent.com/58069494/87297569-f2ac2680-c508-11ea-8e69-021b1a54e86f.png)

- set new data to the chart (series) but only for a way smaller x axis range: 4 minutes (happens also with 4 hours)
![image](https://user-images.githubusercontent.com/58069494/87297522-dad4a280-c508-11ea-849c-32509c77a4e5.png)

**Stack**
```
TypeError: Cannot set property '__disabled' of undefined
    at e.appendDataItems (Component.ts:1048)
    at e.validateDataRange (Component.ts:999)
    at e.validateDataRange (XYSeries.ts:1378)
    at t.update (System.ts:241)
    at System.ts:405
```

**Environment (if applicable)**

 - amCharts version : 4.9.27
 - Browser and its version: All browsers , various versions

**Comment by zeroin** [comment]
Hi,

I think I understand what causes the problem. The issue should be fixed in next release, we will let you know when it happens. Thanks for reporting!

**Comment by martynasma** [comment]
Fixed in [4.9.31](https://github.com/amcharts/amcharts4/releases/tag/4.9.31).

## [4.9.31] - 2020-07-21

### Changed
- Global adapters are being deprecated due to performance reasons. They will still work on some properties, but may be turned off at some future version. If you are using `am4core.globalAdapter`, please consider refactoring your code in some other way.
- If `groupInterval` is manually set, `DateAxis` now prepares all intervals. Normally, the chart does not prepare data sets for time intervals longer than the span of dates in the data. This causes problem in case user sets these intervals manually using `groupInterval`. Note, the groupInterval must be set initially, before data parsing in order data to be grouped into all groupIntervals.

### Fixed
- "Year of week" (`YYYY`) was not being calculated correctly by `DateFormatter`.
- Regression plugin was not working correctly in an all-`ValueAxis` scenarios.
- `ColumnSeries` were not working with `dropFromPreviousState`.
- If data was changed on a chart which was zoomed and there were less data items in the new data than there was before, a JS error was triggered.
- Sometimes a chart with `XYChartScrollbar` with series in it could produce a JS error if series has its `fillOpacity > 0`.
- Series in an `XYChartScrollbar` was not being shown if `options.onlyShowOnViewport = true`.
- `zoomToRectangle()` call was resulting in an error if called after the `MapChart` was rotated.
- `CurvedColumnSeries` columns were not being filled with proper color (black was being used instead).
- A chart with multiple series starting/ending not at the same position could result in wrong zoom on `ValueAxis`.
- `DateAxis` with millisecond `baseInterval` sometimes could show one millisecond less than actual data point in its tooltip.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
